### ISSUE TITLE
Text input between quotes is slow

### ISSUE THREAD

**Issue by f111fei** [VSCodeVim/Vim#1327]
### What did you do?

1. Install `vim` extension.

2. open a large `js` file.  e.g : https://code.jquery.com/jquery-3.1.1.js

3. Typing `I` switch to insert mode.

4. Typing `"` and other letters (e.g: "Hello") at **first line**. The other letters **takes a long time** to appear.

![gif](https://cloud.githubusercontent.com/assets/7069719/23247974/f0ac5f76-f9d7-11e6-97b7-28f5e374e8fa.gif)

### Technical details:

* VSCode Version: 1.9.0
* VsCodeVim Version: 0.5.3
* OS: windows7 and mac

**Comment by johnfn** [comment]
Easy to reproduce. Also very weird.

**Comment by xconverge** [comment]
Yep good catch!

**Comment by xconverge** [comment]
I think that this is definitely a large document vs small document issue... I can't reproduce it in smaller files, it has to be a larger file

**Comment by xconverge** [comment]
Does the same thing happen with snippets? Perhaps the code we are using to make sure cursor position is fine in the case of vscode auto adding text is where this issue lies (ie. iterating whole document or something...)

**Comment by xconverge** [comment]
Ok this is related to the open/close bracket fix that @rufusroflpunch did in #1267

In nextPairedChar, we iterate the whole document. We should change this check the character next to it ONLY instead of first find the pair and THEN compare position.

```
  if (vimState.currentMode === ModeName.Insert) {
      // Check if the keypress is a closing bracket to a corresponding opening bracket right next to it
      let result = PairMatcher.nextPairedChar(vimState.cursorPosition, key, false);
      if (result !== undefined) {
        if (vimState.cursorPosition.compareTo(result) === 0) {
          return true;
        }
      }

      result = PairMatcher.nextPairedChar(vimState.cursorPosition.getLeft(), key, true);
      if (result !== undefined) {
        if (vimState.cursorPosition.getLeftByCount(2).compareTo(result) === 0) {
          return true;
        }
      }
    }
```
